### ISSUE TITLE
Improve behavior when yaml contains a string where a list should be

### ISSUE THREAD

**Issue by chris-martin** [docker/compose#2303]
I just accidentally typed

```
volumes: /data
```

instead of

```
volumes:
 - /data
```

I'd have expected it either to work, or to produce an error message along the lines of "list expected, got string instead". Instead it appears to just blindly iterate over `"/data"`, because the actual output is:

```
System error: mounting into / is prohibited
```

docker-compose version 1.4.2

**Comment by icecrime** [comment]
+1! Ping @dnephin.

**Comment by dnephin** [comment]
Already fixed in 1.5.0!  It will display a proper error message about an invalid type.

You can find a release candidate here: https://github.com/docker/compose/releases/tag/1.5.0rc3
